File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.java
import java.util.*;

class Solution {
    public int[] solution(String[] keymap, String[] targets) {
        HashMap<Character, Integer> shortCut = new HashMap<>();
        
        for (String key: keymap) {
            for (int i = 0; i < key.length(); i++) {
                char c = key.charAt(i);
                if (shortCut.containsKey(c)) {
                    int time = shortCut.get(c);
                    if (time > i + 1) {
                        shortCut.replace(c, i + 1);
                    }
                } else {
                    shortCut.put(c, i + 1);
                }
            }
        }
        
        List<Integer> answer2 = new ArrayList<>();
            for (String target: targets) {
                int key_sum = 0;
                for (char c: target.toCharArray()) {
                    if (shortCut.containsKey(c)) {
                        key_sum += shortCut.get(c);
                    } else {
                        key_sum = -1;
                        break;
                    }
                }
                answer2.add(key_sum);
            }

        int[] answer = answer2.stream()
                .mapToInt(Integer::intValue)
                .toArray();
        return answer;
    }
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    answer = [0 for _ in range(len(targets))]
    
    key_map = dict()
    for i, key in enumerate(keymap):
        for j, v in enumerate(key):
            if v in key_map:
                key_map[v] = [i, min(key_map[v][1], j + 1)]
            else:
                key_map[v] = [i, j + 1]
    
    for i, target in enumerate(targets):
        for v in target:
            if v not in key_map:
                answer[i] = -1
                break
            _, count = key_map[v]
            answer[i] += count
    

    return answer